zerver/tests/test_realm_emoji.py

```diff
@@ -228,10 +246,13 @@ def test_user_settings_for_adding_custom_emoji(self) -> None:
             result = self.client_post("/json/realm/emoji/my_emoji_3", info=emoji_data)
         self.assert_json_success(result)
 
-        do_set_realm_property(
-            othello.realm,
-            "add_custom_emoji_policy",
-            CommonPolicyEnum.MEMBERS_ONLY,
+        members_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.MEMBERS, realm=realm, is_system_group=True
+        )
+        do_change_realm_permission_group_setting(
+            realm,
+            "can_add_custom_emoji_group",
+            members_system_group,
```

Comment:
Would be good to have a test case checking the permission when setting is set to a non-system group and an anonymous group as well.

Reply:
Added tests for that.